src/components/MovieCard/index.tsx

```diff
@@ -1,31 +1,43 @@
 import * as React from 'react';
 import { BsThreeDotsVertical, BsX } from 'react-icons/bs';
 import HandleClickOut from 'Components/HandleClickOut';
-import { useDeleteModal, useEditModal } from '@/hooks/useModal';
-import { useSelectMovie } from '@/hooks/useSelectedMovie';
+import { parseDate } from '@/functions/parseDate';
+import { useDispatch } from 'react-redux';
+import { genresMap } from 'Components/MovieList/helper';
+import { selectMovie } from '@/store/modules/selectedMovie';
+import { openModal } from '@/store/modules/modal';
 import { MovieCardContainer, MovieInfo, MovieOptionsMenu } from './styles';
 import { buildMenuItems } from './helper';
 import { IMovieCardProps } from './types';
 
 const MovieCard: React.FunctionComponent<IMovieCardProps> = ({ movie }) => {
-  const { image, genre, name, year } = movie;
+  const { poster_path, genres, title, release_date, id } = movie;
   const [isMouseOver, setIsMouseOver] = React.useState(false);
   const [isOptionsMenuOpen, setIsOptionsMenuOpen] = React.useState(false);
+  const dispatch = useDispatch();
 
-  const setSelectedMovie = useSelectMovie();
-  const selectMovie = React.useCallback(
-    () => setSelectedMovie(movie),
-    [movie, setSelectedMovie]
+  const chooseMovie = React.useCallback(
+    () => dispatch(selectMovie(movie)),
+    [movie, dispatch]
   );
 
-  const openDeleteModal = useDeleteModal();
-  const openEditModal = useEditModal();
+  const openDeleteModal = React.useCallback(
+    () => dispatch(openModal('delete')),
```

Comment:
Avoid using string literals across your application.
Instead for each module requiring it, create a constants.ts file and assign all your string literals to variables following a pattern like:
export const DELETE = 'delete';
export const GET_MOVIES = 'GET_MOVIES';